src/tests/ftest/util/mdtest_test_base.py

```diff
@@ -44,19 +44,21 @@ def setUp(self):
         self.log.info('Clients %s', self.hostlist_clients)
         self.log.info('Servers %s', self.hostlist_servers)
 
-    def execute_mdtest(self, out_queue=None, display_space=True):
+    def execute_mdtest(self, out_queue=None, display_space=True, cont_namespace=None):
         """Runner method for Mdtest.
 
         Args:
             out_queue (queue, optional): Pass any exceptions in a queue. Defaults to None.
             display_space (bool, optional): Whether to display the pool space. Defaults to True.
+            cont_namespace (str, optional): container namespace when creating a container.
+                Defaults to None.
         """
         # Create a pool if one does not already exist
         if self.pool is None:
             self.add_pool(connect=False)
         # create container
         if self.container is None:
-            self.add_container(self.pool)
+            self.container = self.get_container(self.pool, namespace=cont_namespace)
```

Comment:
Maybe this should be a class-specific version of get_conatianer() - get_mdtest_container() - that ensures that the container is created with the oclass params that match the mdtest params.

Reply:
Agree. Updated